movement/transforms.py

```diff
@@ -113,3 +116,155 @@ def scale(
+    - The computed homography is most accurate for
+        **planar scenes with perspective distortion**,
+        where the transformation can be modeled
+        as a projective mapping.
+
+    Examples
+    --------
+    >>> src = np.array([[0, 0], [1, 0], [1, 1], [0, 1]], dtype=np.float32)
+    >>> dst = np.array([[0, 0], [2, 0], [2, 2], [0, 2]], dtype=np.float32)
+    >>> H = get_homography_transform(src, dst)
+    >>> print(H.shape)
+    (3, 3)
+
+    """
+    srcPoints = np.asarray(srcPoints, dtype=np.float32)
+    dstPoints = np.asarray(dstPoints, dtype=np.float32)
+
+    if len(srcPoints.shape) != 2 or len(dstPoints.shape) != 2:
+        raise ValueError("Points must be 2-dimensional arrays.")
+
+    if srcPoints.shape != dstPoints.shape:
+        raise ValueError(
+            "Source and destination points must have the same shape."
+        )
+
+    dim = srcPoints.shape[1]
+    if dim != 2:
+        raise ValueError("Points must be 2-dimensional.")
+
+    srcPoints, dstPoints = _filter_invalid_points(srcPoints, dstPoints)
+    num_points = srcPoints.shape[0]
+
+    if num_points < 4:
+        raise ValueError(
+            "Insufficient points to compute the homography transformation."
+        )
+
+    transform_matrix, _ = cv2.findHomography(
+        srcPoints, dstPoints, method=cv2.RANSAC
+    )
+
+    return transform_matrix
+
+
+def _filter_invalid_points(
+    src_pts: np.ndarray, dst_pts: np.ndarray, eps: float = 1e-6
+):
+    assert len(src_pts) == len(dst_pts)
+    src_pts = np.asarray(src_pts)
+    dst_pts = np.asarray(dst_pts)
+    keep_idx: list[int] = []
+    obtained_min_non_collinear_set = False
+
+    for i in range(len(src_pts)):
+        # skip duplicates
+        if any(
+            np.linalg.norm(src_pts[i] - src_pts[j]) < eps for j in keep_idx
+        ):
+            continue
+
+        subset = np.vstack([src_pts[j] for j in keep_idx] + [src_pts[i]])
+
+        if subset.shape[0] < 3:
+            keep_idx.append(i)
+            continue
+        elif subset.shape[0] == 3 and _is_collinear_set(subset, eps):
+            continue
+
+        # If we have at least 3 old points, check that
+        # new point is not collinear with any other two
+        if not obtained_min_non_collinear_set and subset.shape[0] > 3:
+            all_noncollinear_triples = all(
+                not _is_collinear_three(
+                    src_pts[a], src_pts[b], src_pts[i], eps
+                )
+                for a, b in itertools.combinations(keep_idx, 2)
+            )
+            if all_noncollinear_triples:
+                obtained_min_non_collinear_set = True
+            else:
+                continue
+        keep_idx.append(i)
+
+    return src_pts[keep_idx], dst_pts[keep_idx]
+
+
+def _is_collinear_three(a, b, c, eps=1e-8):
+    return (
+        abs((b[0] - a[0]) * (c[1] - a[1]) - (b[1] - a[1]) * (c[0] - a[0]))
+        <= eps
+    )
+
+
+def _is_collinear_set(points: np.ndarray, eps: float = 1e-6):
```

Comment:
Similarly, I would add a one-line docstring stating that this uses am SVD-based rank check to determine if a set of points is collinear.